content/en/data_streams/manual_instrumentation.md

```diff
@@ -94,6 +94,69 @@ set_consume_checkpoint(
 "<datastream-type>", "<datastream-name>", headers.get
 )
 
+{{< /code-block >}}
+{{% /tab %}}
+{{% tab ".Net" %}}
+<div class="alert alert-warning">
+  <strong>Note:</strong> DSM instrumentation does not work in Async operations
+</div>
+
+In your producer
+{{< code-block lang="csharp" >}}
+using Datadog.Trace;
+
+using (var scope = Tracer.Instance.StartActive("produce"))
+{
+    var headers = new Headers();
+    var msg = new Message<Null, string> { Value = "any kind of value", Headers = headers};
+
+    new SpanContextInjector().InjectIncludingDsm(
+        msg.Headers,
+        SetHeader,
+        scope.Span.Context,
+        messageType: "<datastream-type>",
+        target: "<queue-or-topic-name>"
+    );
+
+    // Produce the message
+}
+
+// Specific to how the header is modeled
+static void SetHeader(Headers headers, string key, string value)
+{
+    headers.Add(new Header(key, value));
+}
+{{< /code-block >}}
+
+In your consumer
+{{< code-block lang="csharp" >}}
+using Datadog.Trace;
+
+var msg = consumer.Consume();
+var parentContext = new SpanContextExtractor().ExtractIncludingDsm(
+    msg.Headers,
+    GetHeader,
+    messageType: "<datastream-type>",
+    source: "<queue-or-topic-name>"
+);
+var startTime = DateTimeOffset.UtcNow;
```

Comment:
```suggestion

var startTime = DateTimeOffset.UtcNow;
var msg = consumer.Consume();
var parentContext = new SpanContextExtractor().ExtractIncludingDsm(
    msg.Headers,
    GetHeader,
    messageType: "<datastream-type>",
    source: "<queue-or-topic-name>"
);
```
We force a different start time on the span because we want to capture the time it takes to execute the `Consume` operation, so we need to capture it before that.